packages/cache/src/internal/cacheHttpClient.ts

```diff
@@ -149,14 +149,18 @@ export async function reserveCache(
 
   const reserveCacheRequest: ReserveCacheRequest = {
     key,
-    version
+    version,
+    cacheSize: options?.cacheSize
   }
   const response = await retryTypedResponse('reserveCache', async () =>
     httpClient.postJson<ReserveCacheResponse>(
       getCacheApiUrl('caches'),
       reserveCacheRequest
     )
   )
+  if(response?.statusCode === 400){
```

Comment:
As discussed return cacheId and and message from here.

Reply:
Done